crates/rolldown/src/utils/pre_process_ecma_ast.rs

```diff
@@ -49,7 +49,7 @@ impl PreProcessEcmaAst {
         semantic_ret.errors,
         &source,
         path,
-        &Severity::Warning,
+        &Severity::Error,
```

Comment:
If it gonna be put in `warnings`, the `Severity` should be warning.

Reply:
You should return `BuildDiagnostic::from_oxc_diagnostics(..)` as `Err(..)` instead of add them to warnings.